vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -72,6 +78,35 @@
     private Registration validationRegistration;
     private Registration dataProviderListenerRegistration;
 
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(ListDataProvider<T> dataProvider) {
+        this.setDataProvider((DataProvider<T, ?>) dataProvider);
+        return getListDataView();
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(Collection<T> items) {
+        return setDataProvider(DataProvider.ofCollection(items));
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(Stream<T> items) {
+        return setDataProvider(DataProvider.fromStream(items));
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> setDataProvider(T... items) {
+        return setDataProvider(DataProvider.ofItems(items));
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> getListDataView() {
+        if (dataProvider instanceof ListDataProvider) {
```

Comment:
```
public CheckboxGroupListDataView<T> getListDataView() 
```
This could maybe also be added as a default implementation to the `DataView`, if we apply some black magic to where the data provider instance is stored in the components utilizing `ComponentUtil.setData`

Reply:
This needs extra efforts, but that's a good point indeed!